src/hashmap_v2.hpp

```diff
@@ -0,0 +1,177 @@
+            for (size_t i = 0; i < array_size_; i++) {
+                LinkedList* hashmapv2_current_node = hashmapv2.array_[i].next_;
+                LinkedList* current_node           = &array_[i];
+                while (hashmapv2_current_node != nullptr) {
+                    LinkedList* new_node = new LinkedList(hashmapv2_current_node->key_, hashmapv2_current_node->value_);
+                    current_node->next_  = new_node;
+                    current_node         = current_node->next_;
+                    hashmapv2_current_node = hashmapv2_current_node->next_;
+                }
+            }
+        }
+    }
+
+    // Copy Assignment operator
+    HashMapV2& operator=(const HashMapV2& hashmapv2) {
+        destroy();
+        array_size_    = hashmapv2.array_size_;
+        elements_size_ = hashmapv2.elements_size_;
+        if (hashmapv2.array_ != nullptr) {
+            array_ = new LinkedList[array_size_];
+            for (size_t i = 0; i < array_size_; i++) {
+                LinkedList* hashmapv2_current_node = hashmapv2.array_[i].next_;
+                LinkedList* current_node           = &array_[i];
+                while (hashmapv2_current_node != nullptr) {
+                    LinkedList* new_node = new LinkedList(hashmapv2_current_node->key_, hashmapv2_current_node->value_);
+                    current_node->next_  = new_node;
+                    current_node         = current_node->next_;
+                    hashmapv2_current_node = hashmapv2_current_node->next_;
+                }
+            }
+        }
+        return *this;
+    }
+
+    // Move Constructor
+    HashMapV2(HashMapV2&& hashmapv2) : array_size_(hashmapv2.array_size_), elements_size_(hashmapv2.elements_size_) {
+        array_                   = hashmapv2.array_;
+        hashmapv2.array_size_    = 10;
+        hashmapv2.elements_size_ = 0;
+        hashmapv2.array_         = nullptr;
+    }
+
+    // Move Assignment operator
+    HashMapV2& operator=(HashMapV2&& hashmapv2) {
+        destroy();
+        array_size_              = hashmapv2.array_size_;
+        elements_size_           = hashmapv2.elements_size_;
+        array_                   = hashmapv2.array_;
+        hashmapv2.array_size_    = 10;
+        hashmapv2.elements_size_ = 0;
+        hashmapv2.array_         = nullptr;
+        return *this;
+    }
+
+    void destroy() {
+        if (array_ != nullptr) {
+            for (int i = 0; i < array_size_; i++) {
+                LinkedList* current_node = array_[i].next_;
+                while (current_node != nullptr) {
+                    LinkedList* next_node = current_node->next_;
+                    delete current_node;
+                    current_node = next_node;
+                }
+            }
+            delete[] array_;
+        }
+    }
+
+    ~HashMapV2() { destroy(); }
+
+    void construct() { array_ = new LinkedList[array_size_]; }
+
+    // Random hash
+    size_t hash(const K& key) {
```

Comment:
You could use std::hash directly ?

Also, your hash impl is wrong.

Try this:

std::string s1 = "ABC";
std::string s2 = "ABC";

You will get different hash value for s1 and s2.